iceoryx_hoofs/include/iceoryx_hoofs/internal/units/duration.inl

```diff
@@ -423,7 +399,7 @@ inline constexpr bool Duration::wouldCastFromFloatingPointProbablyOverflow(const
     // or the first one which causes an overflow;
     // to be safe, this is handled like causing an overflow which would result in undefined behavior when casting to
     // Seconds_t
-    constexpr To SECONDS_BEFORE_LIKELY_OVERFLOW = static_cast<To>(std::numeric_limits<Seconds_t>::max());
+    constexpr From SECONDS_BEFORE_LIKELY_OVERFLOW = static_cast<From>(std::numeric_limits<Seconds_t>::max());
```

Comment:
@elBoberido patched this for me when I encountered a compiler warning when multiplying duration with a floating point number.

The idea as far as I understood @elBoberido during our whiteboard session is the following.
* In previous methods we ensured that the floatingPoint is always >= 0. So we only need to look at positive floats.
* We cast the max integer M into floating point F. Either it has a value a little bit greater F+ or a little bit smaller F- than the original integer. 
* In between F- and F+ is no value which is representable by that float.
* If the provided floatingPoint is greater than the F we will cause an overflow when we cast the floatingPoint back to integer. 
* If floatingPoint would be equal to F- we could in actually cast floatingPoint into integer but we accept this glitch to make the algorithm easier.

Reply:
I have to think this through but I think @elBoberido should confirm. I understand how floating points work, but I do not understand why suddenly `To` is not used anymore. It should depend on the target type as well in some way.

Maybe not because the internal computation is all done in the source type (`From`)? ... I did not invest much time here but floating point math has the potential for a lot of subtle errors so we need to be careful.

Reply:
@MatthiasKillat because `From` is floating point and `To` is an integer. Previously an `Integer::max` was casted to an integer and the compared with a floating point. This comparison resulted in an implicit cast from the integer to a floating point before the comparison. This comparison would then yield a wrong result since the floating point conversion might already have resulted in a value which is larger that `Integer::max` and the round trip back to an integer would cause an overflow.

Or in other words, we have to cast to the type with the larger range before we do the comparison.

@elfenpiff could you please change `(std::numeric_limits<Seconds_t>::max());` to `(std::numeric_limits<To>::max());`